x-pack/plugins/connectors_networking/server/docs/poc.md

```diff
@@ -0,0 +1,345 @@
+  new plugin makes sense.  It will allow us to largely hide axios as an
+  implementation detail, but more importantly cut down on the amount of
+  cross-talk required between actions and this new plugin.  Probably means
+  also moving networking config properties in actions to the new plugin,
+  so that will require some deprecation.
+
+## trying it out
+
+A bash script is provided in this file:
+
+- `x-pack/plugins/connectors_networking/server/routes/test.sh`
+
+The script creates a webhook that points back into Kibana, posting to a 
+bogus link.  Since running Kibana locally with security on implies you are
+running Kibana with a self-signed certificate, we can use this non-functional
+url to compare running with `rejectAuthenticated` as `true` and `false` to 
+make sure the property is being applied.  When `true` the webhook should fail
+with a TLS self-signed cert error, and when `false` the webhook should fail
+with a plain old 404 error, which implies it made it through the TLS auth.
+
+It also runs a test using the ca that is used to sign the cert that Kibana
+runs with in dev mode.  This ensures that you can actually set a per-server
+ca and have it used on the request.
+
+Enable debug logging via the following in your `kibana.dev.yml`:
+
+```
+logging:
+  loggers:
+    - name: plugins.actions
+      level: debug
+      appenders: [console]
+    - name: plugins.connectorsNetworking
+      level: debug
+      appenders: [console]
+```
+
+## open questions
+
+- Names!  connectorsNetworking?  erghh.  should we use actions instead?
+
+- Do we need to make the saved objects hidden?  I don't think so, but worth
+  asking
+
+- Should these saved objects be global or space-specific or shared?
+
+- Do we need a new privilege to use the CRUD APIs on the saved objects? Or
+  maybe just the mutating APIs?  The lowest privilege to access these would
+  be general access to actions.  It seems like we could use the actions
+  feature control to control access to these saved objects.
+
+- We'll probably want a way for a privileged-enough user creating or editing
+  a connector to add a new CNO right from the connector UI
+
+- Do all our connectors use axios?  Nodemailer is a special case for mail, but
+  I think the only connector not directly using axios is Slack, so perhaps we
+  should stop using Slack's thin shim over axios, and implement it with axios
+  ourselves.  Are there more?
+
+## one more thing - maybe this is just Kibana config?
+
+Since thinking of doing this all via Kibana config, it's kinda stuck in my
+mind.  I suspect it would be good enough for most customers, and a full
+UI treatment probably is over-kill, if customers are likely to need just a
+handful (or less) of these.
+
+If we were to do this, we could follow the lead of preconfigured actions,
+in terms of design/implementation, in preparation if we ever had to do saved
+objects as well (since actions supports both).
+
+Means no UI, no HTTP APIs.  Code is almost done :-)
```

Comment:
Given the complexities for multi-tenant scenarios and the requirement for "protocol:hostname:port"-unique CNOs, I tend to agree with this sentiment. Kibana config will probably satisfy the vast majority of use cases.